src/Stack/Build/Execute.hs

```diff
@@ -896,3 +899,26 @@ getSetupHs dir = do
   where
     fp1 = dir </> $(mkRelFile "Setup.hs")
     fp2 = dir </> $(mkRelFile "Setup.lhs")
+
+-- | Update the symlink in @.stack-work\/install\/last@ to point to the
+-- current local install directory.
+addLastInstallSymlink :: M env m => m ()
+addLastInstallSymlink = do
+  target <- installationRootLocal
+  source <- lastInstallSymlink
+  updateSymlink target source
+
+-- | Update the symlink in @.stack-work\/dist\/last@ to point to the
+-- current local install directory.
+addLastDistSymlink :: M env m => m ()
+addLastDistSymlink = do
+  bc <- asks getBuildConfig
+  target <- distDirFromDir (bcRoot bc)
+  source <- lastInstallSymlink
+  updateSymlink target source
+
+updateSymlink :: M env m => Path Abs Dir -> Path Abs File -> m ()
+updateSymlink target source = do
+  createTree target
+  removeFileIfExists source
+  liftIO $ createSymbolicLink (toFilePath target) (toFilePath source)
```

Comment:
Indeed, I haven't tested this on windows - I wish travis CI supported windows!

Hmm, would the `catchIO` wrapping be for `createSymbolicLink`?  Unless there are some unusual circumstances, that ought to not throw any exceptions, because the target exists and the source doesn't.  Might still be good to wrap it with a more descriptive error message, though?

`createTree` shouldn't ever fail, because it's just a no-op if the tree doesn't exist, and otherwise creates the whole path.  `removeFileIfExists` catches the does-not-exist exception [here](https://github.com/fpco/stack/blob/e9860699de3111dd1d7509f21de8d08d35e86a11/src/Path/IO.hs#L157).